### PR TITLE
reintroduce backward trace

### PR DESCRIPTION
**Pull request by n-hutton**
accidentally removed in commit #3f82faaa

### CODE REVIEW

File: libs/ledger/src/chain/block_coordinator.cpp [hunk 1]
```diff
@@ -206,6 +206,9 @@ BlockCoordinator::BlockCoordinator(MainChain &chain, DAGPtr dag,
   state_machine_->RegisterHandler(State::RESET,                        this, &BlockCoordinator::OnReset);
   // clang-format on
 
+  int *thing = nullptr;
```
Comment (ejfitzgerald) [inline libs/ledger/src/chain/block_coordinator.cpp]: Not sure you want to keep this in

File: scripts/install-test-dependencies.sh [hunk 2]
```diff
@@ -4,3 +4,5 @@ set -e
 pip3 install --no-cache-dir --ignore-installed --force-reinstall wheel pyyaml requests
 pip3 install --no-cache-dir --ignore-installed --force-reinstall fetchai-ledger-api==0.8.0a2
 pip3 install --no-cache-dir --ignore-installed --force-reinstall ./scripts/fetchai_netutils
+
+sudo apt install libdw-dev
```
Comment (ejfitzgerald) [inline scripts/install-test-dependencies.sh]: I think that this should be part of the develop docker image

**Comment by wkaluza** [comment]
Does this complement the changes you made to your fork of `backward-cpp`? They are currently unused on master.

**Comment by n-hutton** [comment]
Yes, this utilizes the fetch logging when stacktracing which is what the fork is for. This was used but you removed it at one point, I presume accidentally. There was also some backtrace macros I will eventually put back in (they can be useful)

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**